### ISSUE TITLE
Move to core

### ISSUE THREAD

**Issue by natefaubion** [doolse/purescript-records#1]
These are super useful, and we need some other things like get/set/add/delete via symbols now that we have RowCons.

/cc @paf31 @garyb

**Comment by paf31** [comment]
I agree we should add something to core. So either:

- We could create something new using the usual template, set up Travis etc., and then maybe @doolse could add this work via PR, or
- @doolse could transfer this one, but we'd need a LICENSE first.

**Comment by natefaubion** [comment]
FWIW, the bower.json lists the license as "MIT", but I don't know if that qualifies :)

**Comment by hdgarrood** [comment]
That should definitely qualify imo. It's not some ad-hoc string, it's a clearly defined language for talking about licenses, namely SPDX. It was because of the existence of SPDX that we decided that `purs publish` should allow packages with no license file and instead check whether the license field was a valid SPDX expression.

**Comment by paf31** [comment]
I don't like to rely on bower.json or package.json for inferring the license, because they're often autogenerated, and the license files do usually say you must recreate the text of the license if you want to reproduce the work.

I don't mind this so much for hosting packages, when someone is doing the uploading anyway, but I do want the license to be explicit for core and contrib packages.

**Comment by hdgarrood** [comment]
Ah okay, that does make a lot of sense actually.

**Comment by doolse** [comment]
I'm happy with whatever you'd like to do. My job has taken me temporarily away from purescript unfortunately, so it might be better if I add one of you as a collaborator and you can do whatever you want to it. As far as the license, yeah MIT was just the default :), i'm happy with that or a similar one.

**Comment by paf31** [comment]
I'll create a repo called `purescript-records` in core with some other useful functions like `get` and `set`, and make it MIT-licensed. Then we can move these functions over in a PR.

_Edit_: actually, that won't work, I didn't realize this repo was registered in Bower already.

**Comment by paf31** [comment]
I could create `purescript-record` without the "s" in core.

**Comment by hdgarrood** [comment]
I think @doolse should be able to update the bower registry as the original register-er of purescript-records..? Not 100% sure of that though.

**Comment by garyb** [comment]
Yep, that's possible - or can remove it from the registry at least, and then re-register the other.

**Comment by coot** [comment]
Playing with `RowList` is wrote a `delete` [function](http://try.purescript.org/?gist=3120e715341dbe38b82e6dbf7a987ab7).

**Comment by doolse** [comment]
@paf31 I tried to de-register on bower but it was failing to log me in for some reason but apparently I can just request a change via the github issue on the bower/registry project if you want? or just go with purescript-record :)

**Comment by paf31** [comment]
Ok, I've created `purescript/purescript-record`. I'll make a PR to add some basic functionality, and please feel free to PR any additional functionality you think would be useful. Thanks!
